packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,157 @@
+
+  // TODO(garyq): We should eventually expand this to install modules by loadingUnitId
+  // as well as moduleName, but currently, loadingUnitId is opaque to the dart code
+  // so this is not possible. The API has been left flexible to allow adding
+  // loadingUnitId as a parameter.
+
+  /// Requests that an assets-only deferred component identified by the [moduleName]
+  /// be downloaded and installed.
+  ///
+  /// This method returns a Future<void> that will complete when the feature is
+  /// installed and any assets are ready to used. When an error occurs, the
+  /// future will complete an error.
+  ///
+  /// This method should be used for asset-only deferred components. Deferred components
+  /// containing dart code should call `loadLibrary()` on a deferred imported
+  /// library's prefix to ensure that the dart code is properly loaded as
+  /// `loadLibrary()` will provide the loading unit id needed for the dart
+  /// library loading process. For example:
+  ///
+  /// ```dart
+  /// import 'split_module.dart' deferred as SplitModule;
+  /// ...
+  /// SplitModule.loadLibrary();
+  /// ```
+  ///
+  /// This method will not load associated dart libraries contained in the dynamic
+  /// feature module, though it will download the files necessary and subsequent
+  /// calls to `loadLibrary()` to load will complete faster.
+  static Future<void> installDeferredComponent({@required String? moduleName}) async {
+    await SystemChannels.deferredComponent.invokeMethod<void>(
+      'installDeferredComponent',
+      <String, dynamic>{ 'loadingUnitId': -1, 'moduleName': moduleName },
+    );
+  }
+  /// Gets the current installation state of the deferred component identified by the
+  /// [moduleName].
+  ///
+  /// This method returns a string that represents the state. Depending on
+  /// the implementation, this string may vary, but the default Google Play
+  /// Store implementation retuns a state in the [DeferredComponentInstallState] enum.
+  ///
+  /// Installations of deferred component modules may be triggered by either calling
+  /// [installDeferredComponent] for assets-only modules or `loadLibrary()` on a deferred
+  /// imported library. Modules not yet requested or do not exist will complete with
+  /// [DeferredComponentInstallState.unknown]. Modules previously installed but not
+  /// loaded in this session will return [DeferredComponentInstallState.installedPendingLoad],
+  /// which indicates that the installtion request call should be repeated to
+  /// complete the loading process.
+  ///
+  /// Due to the async nature of platform channels and network i/o, newly requested
+  /// installs may return null until the installation request has been processed. Thus,
+  /// this state information should be used as purely informational and not as an
+  /// accurate reflection of the readiness of the deferred component. Code and assets
+  /// may only be used after the Future completes succesfully, regardless of what
```

Comment:
"after the Future completes" - That's the future of the `installDeferredComponent` or `loadLibrary` call, right? Not the future that this method returns?